apps/web/src/app/[locale]/challenge/_components/challenge-track-outline.tsx

```diff
@@ -0,0 +1,94 @@
+'use client';
+
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { type ReactNode } from 'react';
+
+import type { getTrackDetails } from '~/app/[locale]/tracks/_components/track.action';
+import { TrackChallenge } from '~/app/[locale]/tracks/_components/track-challenge-card';
+import { TrackProgress } from '~/app/[locale]/tracks/_components/track-progress';
+
+import { Button } from '@repo/ui/components/button';
+import {
+  Sheet,
+  SheetContent,
+  SheetHeader,
+  SheetTitle,
+  SheetTrigger,
+} from '@repo/ui/components/sheet';
+import type { Challenge } from '@repo/db/types';
+import { ChevronRight } from '@repo/ui/icons';
+
+interface Props {
+  children: ReactNode;
+  challenge: Challenge;
+  track: NonNullable<Awaited<ReturnType<typeof getTrackDetails>>>;
+  asChild?: boolean;
+}
+
+export function ChallengeTrackOutline({ children, challenge, track, asChild = false }: Props) {
+  const router = useRouter();
+
+  const trackChallenges = track?.trackChallenges ?? [];
+  const completedTrackChallenges = track.trackChallenges
+    .filter((trackChallenge) => {
+      return (
+        trackChallenge.challenge.submission.length &&
+        trackChallenge.challenge.submission.some((submission) => submission.isSuccessful)
+      );
+    })
+    .map((trackChallenge) => trackChallenge.id);
+  const inProgressTrackChallenges = track.trackChallenges
+    .filter((trackChallenge) => {
+      return (
+        trackChallenge.challenge.submission.length &&
+        trackChallenge.challenge.submission.every((submission) => !submission.isSuccessful)
+      );
+    })
+    .map((trackChallenge) => trackChallenge.id);
+
+  return (
+    <Sheet>
+      <SheetTrigger asChild={asChild}>{children}</SheetTrigger>
+      <SheetContent
+        className="flex w-full flex-col gap-8 overflow-y-scroll sm:max-w-[400px] md:max-w-[540px]"
+        side="left"
+      >
+        <SheetHeader>
+          <SheetTitle>
+            <Button
+              variant="ghost"
+              className="text-foreground flex max-w-full justify-start gap-4 text-lg font-semibold"
+              onClick={() => router.push(`/tracks/${track.id}`)}
```

Comment:
Should the `Button` be a `Link` instead, considering it's ghost?

Reply:
Hey @QuiiBz I'm going to skip this suggestion for now, I see we do this in a handful of places and we can get these in one clean sweep.

I'd love to be able to do the following for a link that visually looks like one of our button variants, but is semantically the correct `<a href>` under the covers:
```
<Button asChild>
  <Link href={}></Link>
</Button>
```
#GoodFirstIssue

I think the latest from `shad/cn` ui includes this. For example, on Kent C Dodds' Epic Stack: https://github.com/epicweb-dev/epic-stack/blob/main/app/components/ui/button.tsx#L46